src/IROperator.h

```diff
@@ -27,21 +28,25 @@ bool is_const(const Expr &e);
 bool is_const(const Expr &e, int64_t v);
 
 /** If an expression is an IntImm or a Broadcast of an IntImm, return
- * a pointer to its value. Otherwise returns nullptr. */
-const int64_t *as_const_int(const Expr &e);
+ * a its value. Otherwise returns std::nullopt. */
+std::optional<int64_t> as_const_int(const Expr &e);
 
 /** If an expression is a UIntImm or a Broadcast of a UIntImm, return
- * a pointer to its value. Otherwise returns nullptr. */
-const uint64_t *as_const_uint(const Expr &e);
+ * its value. Otherwise returns std::nullopt. */
+std::optional<uint64_t> as_const_uint(const Expr &e);
 
 /** If an expression is a FloatImm or a Broadcast of a FloatImm,
- * return a pointer to its value. Otherwise returns nullptr. */
-const double *as_const_float(const Expr &e);
+ * return its value. Otherwise returns std::nullopt. */
+std::optional<double> as_const_float(const Expr &e);
 
-/** Is the expression a constant integer power of two. Also returns
- * log base two of the expression if it is. Only returns true for
- * integer types. */
-bool is_const_power_of_two_integer(const Expr &e, int *bits);
+/** Is the expression a constant integer power of two. Returns log base two of
+ * the expression if it is, or std::nullopt if not. Also returns std::nullopt
+ * for non-integer types. */
+// @{
+std::optional<int> is_const_power_of_two_integer(const Expr &e);
```

Comment:
style nit: `is_foo` implies the function returns bool, which these no longer do. probably `as_const_foo` is now a better name for these.

Reply:
The issue is that it doesn't return a const power of two. It returns the number of bits in it, if it is one. So "as" is actively misleading. I couldn't immediately think of a better name. If you treat the return value as a bool, "is" still makes sense. Open to other suggestions. exact_integer_log_2?

Reply:
Ah, ok then... name is fine I guess